### PR TITLE
Update to Fastify v4

### PR DESCRIPTION
**Pull request by mcollina**
Signed-off-by: Matteo Collina <hello@matteocollina.com>

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -143,12 +144,7 @@ fastify.get('/', { websocket: true }, function wsHandler (connection, req) {
   })
 })
 
-fastify.listen(3000, err => {
-  if (err) {
-    fastify.log.error(err)
-    process.exit(1)
-  }
-})
+await fastify.listen(3000)
```
Comment (jsumners) [inline README.md:147]:
```suggestion
await fastify.listen({ port: 3000 })
```

File: README.md [hunk 2]
```diff
@@ -127,11 +129,10 @@ This plugin uses the same router as the `fastify` instance, this has a few impli
...
 
 ```js
-'use strict'
-
-const fastify = require('fastify')()
+import Fastify from 'fastify'
 
-fastify.register(require('@fastify/websocket'))
+const fastify = Fastify()
+await fastify.register(require('@fastify/websocket'))
```
Comment (airhorns) [inline README.md:135]: Maybe more consistent to use an es import for this instead of mixing styles?

File: index.js [hunk 3]
```diff
@@ -92,21 +93,30 @@ function fastifyWebsocket (fastify, opts, next) {
...
+  })
+
   fastify.addHook('onRoute', routeOptions => {
     let isWebsocketRoute = false
     let wsHandler = routeOptions.wsHandler
     let handler = routeOptions.handler
 
     if (routeOptions.websocket || routeOptions.wsHandler) {
-      if (routeOptions.method !== 'GET') {
+      if (routeOptions.method === 'HEAD') {
```
Comment (airhorns) [inline index.js:109]: Should we add a test for this?

File: index.js [hunk 4]
```diff
@@ -92,21 +93,30 @@ function fastifyWebsocket (fastify, opts, next) {
     done()
   })
 
+  fastify.addHook('onResponse', (request, reply, error, done) => {
+    if (request.ws) {
+      request.raw[kWs].destroy()
+    }
+    done()
+  })
```
Comment (airhorns) [inline index.js:101]: Worth writing a regression test for this? Is probably quite annoying, I am ok without

**Review by airhorns — APPROVED** [review]
LGTM after tests

File: index.js (hunk 3 above)
Comment (mcollina) [reply index.js:109]: It's covered already!

File: README.md (hunk 2 above)
Comment (mcollina) [reply README.md:135]: I did the best of showing both stiles, either with with `await app.register()` or with using a plugin.

File: index.js (hunk 4 above)
Comment (mcollina) [reply index.js:101]: Without this change, the test would fail. Truth be told, I didn't know why they were passing before.

Comment (airhorns) [reply index.js:101]: Oh, strange! But ok!

**Pull request merged**